.github/workflows/maven.yaml

```diff
@@ -0,0 +1,40 @@
+name: Publish icu4j.jar/utilities.jar to GH Maven
+on:
+  release:
+    types: [created]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: true
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: icu4j ant releaseCLDR
+        run: ant -noinput releaseCLDR -f icu4j/build.xml
+      - name: deploy it
+        run: |
+          MYVERSION=$(echo ${GITHUB_REF} | cut -d/ -f3 | tr . _)
+          echo Github Ref ${GITHUB_REF} @ ${GITHUB_SHA}, version ${MYVERSION}
+          mvn deploy:deploy-file -DgroupId=com.ibm.icu \
+            -DartifactId=icu4j \
+            -Dversion=0.0.0-${MYVERSION} \
```

Comment:
Based on the intention described in the [Jira comment](https://unicode-org.atlassian.net/browse/ICU-21251?focusedCommentId=157956), I think the ["typical Maven way to do it"](https://maven.apache.org/guides/getting-started/index.html#What_is_a_SNAPSHOT_version) is for the development version number to be "<semver>-SNAPSHOT", and the release artifact version number is <semver>.

FWIW, in a previous workplace which used git, one of the teams introduced the git hash into the Maven artifact version numbers, which required downstream teams to deal with this convention.  Many of the senior developers found the benefits to be minimal and outweighed by the hassle caused during development -- every time they made a new git commit locally and wanted to re-upload the resulting binary to test out, they had to manually find the new binary file name and upload, and their build output directory would be cluttered with several development snapshot build files.  They grumbled that the git hash version numbering came into existence as a band-aid solution to immediate problems instead of properly solving the root cause by decomplecting a complex spaghetti architecture.  That example is much more extreme than what is going on here, though, so not sure how relevant that feels.

Reply:
(^ I added some backticks to your comment)
> `<semver…>` 

Yes, but there's no semver that makes a lot of sense here. It really is an arbitrary snapshot. We could make a completely unrelated semver just for this purpose I suppose.

> git hash into the Maven artifact version numbers, which required downstream teams to deal with this convention

in this case, the same person (such as @pedberg-icu  ) is cutting the ICU4J version on the ICU side… and referencing it on the CLDR side. A git hash could work.

Maybe a structured tag could make some sense.  I assume the 'on: release: … ` can take a regex or glob? Then on the ICU side we create releases named:`cldr/38.0.0` or `cldr/38.0.1` etc…  and title it "snapshot 0 for CLDR 38…". 

Then, this workflow can pickup the maven version from there:  "38.0.0-SNAPSHOT" with the "cldr" classifier. I think that will make it not interfere with the regular artifacts, even if we had CLDR v68. 

ICU folks, any thoughts about the tag naming structure?